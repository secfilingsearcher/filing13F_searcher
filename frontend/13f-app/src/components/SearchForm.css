.form {float: 'left', padding: '20px', margin: '20px'}