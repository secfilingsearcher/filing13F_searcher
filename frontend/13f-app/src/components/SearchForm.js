import React, { useRef, useState } from 'react'
import { useFormik } from 'formik'
import { useHistory } from 'react-router-dom'
import { InputGroup, Form, Button, Overlay, Tooltip } from 'react-bootstrap'
import * as Yup from 'yup'

const validationSchema = Yup.object().shape({
  searchName: Yup.string().required()
})

const SearchForm = () => {
  const history = useHistory()
  const [show, setShow] = useState(false)
  const target = useRef(null)
  const showTooltip = () => {
    if (formik.errors.searchName) {
      setShow(false)
    } else {
      setShow(true)
    }
  }
  const formik = useFormik({
    initialValues: {
      searchName: ''
    },
    validationSchema,
    onSubmit: (values, { setSubmitting }) => {
      const searchLink = `/search?q=${values.searchName}`
      history.push(searchLink, { replace: true })
      setSubmitting(true)
      if (formik.errors.searchName) { setShow(!show) }
    }
  })
  return (
       <Form onSubmit={formik.handleSubmit}>
         <InputGroup>
          <Form.Control
            id="searchName"
            name="searchName"
            type="text"
            onChange={formik.handleChange}
            value={formik.values.searchName}
            placeholder="Company Name"
            ref={target}
          />
          <Overlay target={target.current} show={show} placement='bottom'>
            <Tooltip>Required</Tooltip>
          </Overlay>
       <Button type="submit" onClick={showTooltip}><i className="bi bi-search"></i></Button>
       </InputGroup>
    </Form>
  )
}

export default SearchForm
