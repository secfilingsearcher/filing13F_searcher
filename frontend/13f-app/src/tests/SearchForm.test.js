import React from 'react'
import renderer from 'react-test-renderer'
import SearchForm from '../components/SearchForm'

test('SearchForm renders', ()=> {
    // October 20th, 2021
  const date = (new Date(2021, 10, 20)).toISOString().substr(0, 10)
  const component = renderer.create(
        <SearchForm startDate={date} endDate={date}/>
  )
let tree = component.toJSON()
expect(tree).toMatchSnapshot()
})
