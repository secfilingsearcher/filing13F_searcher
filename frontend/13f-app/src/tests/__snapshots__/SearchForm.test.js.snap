// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      id="searchName"
      name="searchName"
      onChange={[Function]}
      placeholder="Company Name"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      <i
        className="bi bi-search"
      />
    </button>
  </div>
</form>
`;
